### PR TITLE
Use Post for Streaming Endpoints

### PR DESCRIPTION
**Pull request by NarekA**
Http streaming breaks when the input doc schema has fields which are not str, int, or float. This includes dicts, bools, and nested objects (See [Issue 6090](https://github.com/jina-ai/jina/issues/6090)). In addition it caps the input size to 2000 characters (Much less when you factor in URL encoding)

This PR changes the get endpoints to post endpoints. Not sure if we want to keep the get endpoints for backwards compatibility.

**Goals:**

- resolves [Issue 6090](https://github.com/jina-ai/jina/issues/6090)
- [x] check and update documentation. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#-contributing-documentation) and ask the team.

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
Can't we keep a GET endpoint together with a POST endpoint, and have the Jina client use the POST? And then we would need to document the limitations of the GET endpoint

File: jina/serve/runtimes/worker/http_fastapi_app.py [hunk 1]
```diff
@@ -130,20 +130,19 @@ def add_streaming_get_route(
 
         @app.api_route(
             path=f'/{endpoint_path.strip("/")}',
-            methods=['GET'],
+            methods=['POST'],
```
Comment (alaeddine-13) [inline jina/serve/runtimes/worker/http_fastapi_app.py:133]: I would rather duplicate the streaming endpoint in order to have 2 versions, post and get. The jina client should use the post method

Comment (NarekA) [reply jina/serve/runtimes/worker/http_fastapi_app.py]: Sounds good to me, will add the get endpoint back.

**Comment by NarekA** [comment]
Added the get endpoints back, and I will add some tests for the post endpoint. This feature will also align us with openai since they are using [streaming on post endpoints](https://platform.openai.com/docs/api-reference/chat/create) too.

**Review by JoanFM — CHANGES_REQUESTED** [review]
Hey @NarekA,

This looks quite cool, I woule be missing a test proving that the original issue is solved. You can add it under the '/tests/integration/docarray_v2' folder.

**Comment by NarekA** [comment]
@JoanFM @alaeddine-13 created a new [PR](https://github.com/jina-ai/jina/pull/6093) that follows conventions, working on adding tests too.

**Pull request closed without merging**